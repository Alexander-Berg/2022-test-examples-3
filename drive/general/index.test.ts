import { prepareKeyToMDS } from './prepareKeyToMDS';

const PATH_TO_FILE = 'path/to/file.png';
const SEPARATE_INDEX = 5;
describe('Prepare key to MDS', () => {
    it('wrong first slash', () => {
        const key = prepareKeyToMDS(`/${PATH_TO_FILE}`);
        expect(key).toEqual(PATH_TO_FILE);
    });

    it('clear spaces in path', () => {
        const key = prepareKeyToMDS(`${PATH_TO_FILE.split('').join(' ')}`);
        expect(key).toEqual(PATH_TO_FILE);
    });

    it('check combo', () => {
        const key = prepareKeyToMDS(`/${PATH_TO_FILE.substr(0, SEPARATE_INDEX)}  `
            + ` ${PATH_TO_FILE.substr(SEPARATE_INDEX)}`);
        expect(key).toEqual(PATH_TO_FILE);
    });
});
