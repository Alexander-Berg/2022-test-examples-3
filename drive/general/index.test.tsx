import React from 'react';
import { render } from '@testing-library/react';

import DashboardScoringLabel from 'components/ui/DashboardScoringLabel/index';

describe('ScoringLabel test', () => {
    describe('ScoringLabel title', () => {
        const MOCK_SCORE = 3;

        it('Should match snapshot with default title', () => {
            let { baseElement } = render(<DashboardScoringLabel score={MOCK_SCORE} />);
            expect(baseElement).toMatchSnapshot();
        });

        it('Should match snapshot with custom title', () => {
            let { baseElement } = render(
                <DashboardScoringLabel
                    title="custom title"
                    score={MOCK_SCORE}
                />,
            );
            expect(baseElement).toMatchSnapshot();
        });
    });

    describe('ScoringLabel score', () => {
        it('Should match snapshot with zero score', () => {
            let { baseElement } = render(<DashboardScoringLabel score={0} />);
            expect(baseElement).toMatchSnapshot();
        });

        it('Should match snapshot with bad score', () => {
            let { baseElement } = render(<DashboardScoringLabel score={1} />);
            expect(baseElement).toMatchSnapshot();
        });

        it('Should match snapshot with normal score', () => {
            let { baseElement } = render(<DashboardScoringLabel score={3.5} />);
            expect(baseElement).toMatchSnapshot();
        });

        it('Should match snapshot with good score', () => {
            let { baseElement } = render(<DashboardScoringLabel score={4.5} />);
            expect(baseElement).toMatchSnapshot();
        });

        it('Should match snapshot with three-digits score', () => {
            let { baseElement } = render(<DashboardScoringLabel score={4.123} />);
            expect(baseElement).toMatchSnapshot();
        });
    });
});
