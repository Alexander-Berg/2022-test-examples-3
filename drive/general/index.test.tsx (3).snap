// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Info Tests Should render UserCardLinks 1`] = `
<div>
  <h4>
    Карточка пользователя
  </h4>
  <div>
    <Link
      href="#/clients/70cea50e-e079-45c3-996c-d65bcb8e7dd2/info"
      target="_blank"
    >
      КЦ
    </Link>
    <span>
       • 
    </span>
    <Link
      href="#/clients/70cea50e-e079-45c3-996c-d65bcb8e7dd2/sessions"
      target="_blank"
    >
      Сессии
    </Link>
    <span>
       • 
    </span>
    <Link
      href="#/clients/70cea50e-e079-45c3-996c-d65bcb8e7dd2/bonuses"
      target="_blank"
    >
      Бонусы
    </Link>
    <span>
       • 
    </span>
    <Link
      href="#/clients/70cea50e-e079-45c3-996c-d65bcb8e7dd2/billing"
      target="_blank"
    >
      Биллинг
    </Link>
    <span>
       • 
    </span>
    <Link
      href="#/clients/70cea50e-e079-45c3-996c-d65bcb8e7dd2/info-old"
      target="_blank"
    >
      Обзор
    </Link>
  </div>
</div>
`;

exports[`Chat Info Tests Should render UserCardLinks with empty title 1`] = `
<div>
  <h4 />
  <div>
    <Link
      href="#/clients/70cea50e-e079-45c3-996c-d65bcb8e7dd2/info"
      target="_blank"
    >
      КЦ
    </Link>
    <span>
       • 
    </span>
    <Link
      href="#/clients/70cea50e-e079-45c3-996c-d65bcb8e7dd2/sessions"
      target="_blank"
    >
      Сессии
    </Link>
    <span>
       • 
    </span>
    <Link
      href="#/clients/70cea50e-e079-45c3-996c-d65bcb8e7dd2/bonuses"
      target="_blank"
    >
      Бонусы
    </Link>
    <span>
       • 
    </span>
    <Link
      href="#/clients/70cea50e-e079-45c3-996c-d65bcb8e7dd2/billing"
      target="_blank"
    >
      Биллинг
    </Link>
    <span>
       • 
    </span>
    <Link
      href="#/clients/70cea50e-e079-45c3-996c-d65bcb8e7dd2/info-old"
      target="_blank"
    >
      Обзор
    </Link>
  </div>
</div>
`;

exports[`Chat Info Tests Should render UserCardLinks with empty userId 1`] = `
<div>
  <h4>
    Карточка пользователя
  </h4>
  <div>
    <Link
      href="#/clients//info"
      target="_blank"
    >
      КЦ
    </Link>
    <span>
       • 
    </span>
    <Link
      href="#/clients//sessions"
      target="_blank"
    >
      Сессии
    </Link>
    <span>
       • 
    </span>
    <Link
      href="#/clients//bonuses"
      target="_blank"
    >
      Бонусы
    </Link>
    <span>
       • 
    </span>
    <Link
      href="#/clients//billing"
      target="_blank"
    >
      Биллинг
    </Link>
    <span>
       • 
    </span>
    <Link
      href="#/clients//info-old"
      target="_blank"
    >
      Обзор
    </Link>
  </div>
</div>
`;
